charts/rancher-monitoring/101.1.0+up40.1.2/charts/grafana/templates/servicemonitor.yaml

```diff
@@ -30,6 +30,22 @@ spec:
     tlsConfig:
     {{- toYaml .Values.serviceMonitor.tlsConfig | nindent 6 }}
     {{- end }}
+    {{- if .Values.serviceMonitor.metricRelabelings }}
+    metricRelabelings:
+    {{- toYaml .Values.serviceMonitor.metricRelabelings | nindent 6 }}
+    {{ if .Values.global.cattle.clusterId }}
+    - sourceLabels: [__address__]
+      targetLabel: cluster_id
+      replacement: {{ .Values.global.cattle.clusterId }}
+    {{- end }}
+    {{ else }}
+    {{ if .Values.global.cattle.clusterId }}
+    metricRelabelings:
+    - sourceLabels: [__address__]
+      targetLabel: cluster_id
+      replacement: {{ .Values.global.cattle.clusterId }}
+    {{- end }}
+    {{- end }}
```

Comment:
```suggestion
    metricRelabelings:
    {{- if .Values.serviceMonitor.metricRelabelings }}
    {{- toYaml .Values.serviceMonitor.metricRelabelings | nindent 6 }}
    {{- end }}
    {{ if .Values.global.cattle.clusterId }}
    - sourceLabels: [__address__]
      targetLabel: cluster_id
      replacement: {{ .Values.global.cattle.clusterId }}
    {{- end }}
```

Reply:
If we're always going to add `metricRelabelings`, we should just remove it from the if block.

Reply:
This should be fixed in all non-pushprox charts.

Reply:
some of the charts have values for metricRelabelings in the values.yaml file, but not all of them do so that was the primary reason for the check 

the other reason is in the comments here: 
https://github.com/rancher/charts/pull/2233#discussion_r1048843829